lms/tasks/roster.py

```diff
@@ -37,6 +38,13 @@ def schedule_fetching_course_rosters() -> None:
     # We use the python version (and not func.now()) for easier mocking during tests
     now = datetime.now()
 
+    # Only fetch roster for courses for which we haven't schedule a fetch recently
+    no_recent_scheduled_roster_fetch_clause = ~exists(
+        select(TaskDone).where(
+            TaskDone.key == func.concat("roster::course::", LMSCourse.id),
+        )
+    )
+
     # Only fetch roster for courses that don't have recent roster information
     no_recent_roster_clause = ~exists(
```

Comment:
We can't remove the existing `no_recent_roster_clause` because all recent roster won't have the corresponding `task_done`.